### PR TITLE
docs(opentelemetry source): Refresh documentation and examples

### PR DESCRIPTION
**Pull request by spencergilbert**
Part of https://github.com/vectordotdev/vector/issues/13884, ~~unsure about the `examples` portion for the `source` - I can provide it as JSON-mapped proto, but that's not how an event could be submitted to the source.~~

EDIT: discussed the examples problem with @jszwedko and we added a new cue field for example configurations that aren't used today, but are more meaningful than the inputs/config/outputs field that existed today.

This drops the `description` field from the CUE docs since it was a 1:1 copy of the one in the markdown front matter.

https://deploy-preview-13921--vector-project.netlify.app/docs/reference/configuration/sources/opentelemetry/

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
👍 nice work. These descriptions and config option examples are much better now.

I thought a bit more about the config examples, and I think we should separate them from the input/output examples. My thought is:

* It's difficult to show the input for some sources (like otel and kafka)
* For some input/output examples we may want to use the same component config but different inputs to show different output

I'll work on a PR refactoring the cue schema here.

**Comment by jszwedko** [comment]
> I'll work on a PR refactoring the cue schema here.

https://github.com/vectordotdev/vector/pull/13928

File: website/cue/reference/components/sources/opentelemetry.cue [hunk 1]
```diff
@@ -98,6 +98,28 @@ components: sources: opentelemetry: {
 		}
 	}
 
+	configuration_examples: [
```
Comment (jszwedko) [inline website/cue/reference/components/sources/opentelemetry.cue:101]: 👍

File: website/cue/reference/components/sources/opentelemetry.cue [hunk 2]
```diff
@@ -98,6 +98,28 @@ components: sources: opentelemetry: {
...
+		{
+			title: "OTLP Defaults"
+			configuration: {
+				opentelemetry: {
+					grpc: {
+						address: "0.0.0.0:4317"
+						tls: {
+							enabled: true
```
Comment (jszwedko) [inline website/cue/reference/components/sources/opentelemetry.cue:109]: If we are enabling TLS don't we need to provide the certificate and key to use?

File: website/cue/reference/components/sources/opentelemetry.cue [hunk 3]
```diff
@@ -98,6 +98,28 @@ components: sources: opentelemetry: {
 		}
 	}
 
+	configuration_examples: [
+		{
+			title: "OTLP Defaults"
```
Comment (jszwedko) [inline website/cue/reference/components/sources/opentelemetry.cue:103]: Is TLS the default? I could see it being unneeded on localhost.

**Review by jszwedko — COMMENTED** [review]
Looks good! Just one question below.

File: website/cue/reference/components/sources/opentelemetry.cue (hunk 3 above)
Comment (spencergilbert) [reply website/cue/reference/components/sources/opentelemetry.cue:103]: TLS is enabled by default on the otel-col: https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md#tls--mtls-configuration

Comment (spencergilbert) [reply website/cue/reference/components/sources/opentelemetry.cue:103]: Added in 18a095957

File: website/cue/reference/components/sources/opentelemetry.cue (hunk 2 above)
Comment (spencergilbert) [reply website/cue/reference/components/sources/opentelemetry.cue:109]: Added in 18a095957

File: website/cue/reference/components/sources/opentelemetry.cue [hunk 4]
```diff
@@ -98,6 +98,34 @@ components: sources: opentelemetry: {
...
+			configuration: {
+				opentelemetry: {
+					grpc: {
+						address: "0.0.0.0:4317"
+						tls: {
+							enabled:  true
+							ca_file:  "/var/vector/ca.pem"
```
Comment (jszwedko) [inline website/cue/reference/components/sources/opentelemetry.cue:110]: The `ca_file` shouldn't be needed unless they want to do client certificate verification (not super common in my experience).

File: website/cue/reference/components/sources/opentelemetry.cue [hunk 5]
```diff
@@ -98,6 +98,34 @@ components: sources: opentelemetry: {
...
+							crt_file: "/var/vector/cert.pem"
+							key_file: "/var/vector/cert.key"
```
Comment (jszwedko) [inline website/cue/reference/components/sources/opentelemetry.cue:112]:
```suggestion
							crt_file: "/etc/ssl/certs/vector.pem"
							key_file: "/etc/ssl/private/vector.key"
```

If we want to give more realistic real-world example.

**Review by jszwedko — APPROVED** [review]
A couple of suggestions below. Otherwise 🎉

File: website/cue/reference/components/sources/opentelemetry.cue (hunk 4 above)
Comment (jszwedko) [reply website/cue/reference/components/sources/opentelemetry.cue:110]: Ah, gotcha, if it is common in OTel land then makes sense to include. I'd suggest `/etc/ssl/certs/vector_ca.pem`

Comment (spencergilbert) [reply website/cue/reference/components/sources/opentelemetry.cue:110]: Also just of the thoughts of "more example = better"

**Pull request merged**